### PR TITLE
Vert.x upgrade

### PR DESCRIPTION
**Pull request by jflo**
## PR description

Upgrades vert.x dependency. 

Notable areas of impact:

- Getting connections from an HTTP client is in itself asynchronous, so another layer of futures is added.
- The way keypairs are added to the JWTAuthProvider is different, and now they are configured using PEM strings.
- JWT attributes and principal fields are changed, for example exp (expiry) is a user attribute, not a principal property.

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
 import org.hyperledger.besu.testutil.TestClock;
 import org.hyperledger.enclave.testutil.EnclaveKeyConfiguration;
 import org.hyperledger.enclave.testutil.EnclaveTestHarness;
-import org.hyperledger.enclave.testutil.OrionTestHarnessFactory;
+import org.hyperledger.enclave.testutil.TesseraTestHarnessFactory;
```
Comment (fab-10) [inline besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java:66]: Why this change from Orino to Tessera?

File: enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java [hunk 2]
```diff
@@ -75,24 +77,34 @@ public VertxRequestTransmitter(final HttpClient httpClient) {
...
+      options.setMethod(method);
+      options.setURI(endpoint);
       if (withAcceptJsonHeader) {
         // this is needed when using Tessera GET /transaction/{hash} to choose the right RPC
-        request.putHeader(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
+        options.putHeader(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
       }
-      contentType.ifPresent(ct -> request.putHeader(HttpHeaders.CONTENT_TYPE, ct));
+      contentType.ifPresent(ct -> options.putHeader(HttpHeaders.CONTENT_TYPE, ct));
+
```
Comment (fab-10) [inline enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java:89]: just a consideration, about the new approach of Vert.x 4 that is harder to read and more verbose that the previous one :(

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java [hunk 3]
```diff
@@ -285,19 +286,19 @@ private void handleGraphQLRequest(final RoutingContext routingContext) {
       final Map<String, Object> variables;
       final HttpServerRequest request = routingContext.request();
 
-      switch (request.method()) {
-        case GET:
+      switch (request.method().name()) {
```
Comment (fab-10) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java:289]: why switching to string when enum is safer?

File: besu/build.gradle [hunk 4]
```diff
@@ -94,4 +94,5 @@ dependencies {
   testImplementation 'org.awaitility:awaitility'
   testImplementation 'org.mockito:mockito-core'
   testImplementation 'commons-io:commons-io'
+  testImplementation 'org.testcontainers:testcontainers'
```
Comment (fab-10) [inline besu/build.gradle:97]: where is this used?

**Review by fab-10 — COMMENTED** [review]
mostrly question to better understand the changes

File: besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java (hunk 1 above)
Comment (jflo) [reply besu/src/test/java/org/hyperledger/besu/PrivacyReorgTest.java:66]: Orion is end-of-life and Tessera is the new replacement. This was easier than updating the Orion harness.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java (hunk 3 above)
Comment (jflo) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java:289]: Sadly, they are not Enums, but rather static instances of HttpMethod :(

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java [hunk 5]
```diff
@@ -213,7 +213,7 @@ private void validateConfig(final GraphQLConfiguration config) {
   }
 
   private Optional<String> getAndValidateHostHeader(final RoutingContext event) {
-    final Iterable<String> splitHostHeader = Splitter.on(':').split(event.request().host());
+    final Iterable<String> splitHostHeader = Splitter.on(':').split(event.request().getHeader(HttpHeaders.HOST));
```
Comment (jflo) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java]: This is an optional header, I think we need some null checking to do this safely, otherwise we have test regressions.

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/pubsub/WebSocketConnection.java [hunk 6]
```diff
@@ -85,30 +85,33 @@ private JsonRpcSuccessEvent send(final String json) {
   private void connect(final Vertx vertx) {
     vertx
         .createHttpClient(new HttpClientOptions())
-        .websocket(
+        .webSocket(
```
Comment (jflo) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/pubsub/WebSocketConnection.java:88]: Lots of things named websocket, inconsistently, are now webSocket everywhere.

File: besu/build.gradle (hunk 4 above)
Comment (jflo) [reply besu/build.gradle:97]: it was introduced via the TesseraTestHarness.

File: besu/src/test/java/org/hyperledger/besu/RunnerTest.java [hunk 7]
```diff
@@ -378,8 +380,6 @@ private void syncFromGenesis(final SyncMode mode, final GenesisConfigFile genesi
         runnerBehind.close();
         runnerBehind.awaitStop();
       }
-      runnerAhead.close();
```
Comment (jflo) [inline besu/src/test/java/org/hyperledger/besu/RunnerTest.java:381]: Ok this one was a huge pain to troubleshoot. Closing runnderBehind closes its vertx, which is shared with runnerAhead. Attempting to close an already closed vertx throws an exception.

File: consensus/clique/build.gradle [hunk 8]
```diff
@@ -46,6 +46,7 @@ dependencies {
 
   implementation 'com.google.guava:guava'
   implementation 'io.vertx:vertx-core'
+  implementation 'com.fasterxml.jackson.core:jackson-databind'
```
Comment (jflo) [inline consensus/clique/build.gradle:49]: Previously provided transitively.

File: enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java (hunk 2 above)
Comment (jflo) [reply enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java:89]: Using the WebClient instead of HttpClient would be more readable, but I was wary of making a more dramatic change for the upgrade.

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/pubsub/WebSocketConnection.java (hunk 6 above)
Comment (macfarla) [reply acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/pubsub/WebSocketConnection.java:88]: ++ consistency

**Comment by usmansaleem** [comment]
@jflo https://github.com/hyperledger/besu/pull/3106 should sort out EthSigner dependency.

**Comment by jflo** [comment]
closing in favor of #3135 , thanks @diega
